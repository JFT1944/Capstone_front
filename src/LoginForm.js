import React, { useState } from "react";
import { useNavigate } from 'react-router-dom'
import DispensaApi from "./ApiHelper";


// This component allows the user to login.
// it also sets the state, updates the joblyapi, and the localstorage 
// with the user info and token.
function LoginForm(props){
let {user, setUser, token, setToken} = props
const navigate = useNavigate();
// holds form information
const [formData, getFormData] = useState({
    username: '', 
    password: ''
})
// let {tokenState, token, user, userState} = props
// console.log(token)


const handleChange = (evt) =>{

    // updatedFormData(evt.target.value)
    const {name, value} = evt.target;
    getFormData(data => ({
        ...data,
        [name]: value
    }))
}
// handles the submit of the form
function clickHandler(e){
    e.preventDefault()
    if(!formData.username || !formData.password){
        alert('please input correct information')
        return
    }
    // communicates with the backend to login
    DispensaApi.login(formData.username, formData.password).then((res) => {
        console.log(res)
       try {
        setUser(res.user)
        setToken(res.token)
        localStorage.setItem('token', res.token)
        localStorage.setItem('user', JSON.stringify(res.user))
       } catch (error) {
        console.log(error)
       }


        navigate('/')
    })
    // DispensaApi.login(formData).then((res) => {
    
    //     // console.log(res)
    //     tokenState(res)
    //     navigate('/')
        // console.log(formData)
        // DispensaApi.getUser(formData.username, res[0]).then((resx) => {
        //     let {email, firstName, lastName, username} = formData
        //     // console.log({resx:resx})
        //     userState(resx)
        //     localStorage.setItem('userInfo', JSON.stringify(resx))
        // })

    // })

    // console.log(user)

}

    return(
        <div style={{display:'flex', justifyContent:'center', flexDirection:'column', alignContent:'center',alignItems:'center'}}>
            <h2>Login!</h2>
            {/* <form className={'iForms'} style={{display:'flex', flexDirection:'column', padding:'20px', maxWidth:'500px', minWidth:'300px', alignContent:'center', justifyContent:'center'}}> */}
            <form className={'iForms'} >
                <span><input className='input_box' type='input' name='username' placeholder="Username" value={formData.username} onChange={(e) => handleChange(e)}></input></span>
                <span><input className='input_box' type='input' name ='password' placeholder="Password" value={formData.password} onChange={(e) => handleChange(e)}></input></span>
                <span className="submit_button"><input type='submit' onClick={(e) => clickHandler(e)} data-testid='subBut'></input></span>
            </form>
        </div>
    )
}


export default LoginForm