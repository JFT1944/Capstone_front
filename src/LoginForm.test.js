
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { BrowserRouter, MemoryRouter } from "react-router-dom";




it('should appear in a smoketest', () => {
    <BrowserRouter>
    render(<LoginForm user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/>)
    </BrowserRouter>
})


it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><LoginForm user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})
