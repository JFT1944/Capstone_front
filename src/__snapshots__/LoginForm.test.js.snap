// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: center; flex-direction: column; align-content: center; align-items: center;"
  >
    <h2>
      Login!
    </h2>
    <form
      class="iForms"
    >
      <span>
        <input
          class="input_box"
          name="username"
          placeholder="Username"
          type="input"
          value=""
        />
      </span>
      <span>
        <input
          class="input_box"
          name="password"
          placeholder="Password"
          type="input"
          value=""
        />
      </span>
      <span
        class="submit_button"
      >
        <input
          data-testid="subBut"
          type="submit"
        />
      </span>
    </form>
  </div>
</DocumentFragment>
`;
